src/components/LoadingModal.vue

```diff
@@ -0,0 +1,242 @@
+        <!-- 캐릭터들이 일렬로 배치 -->
+        <div
+          v-for="(character, index) in characters"
+          :key="character.name"
+          class="flex flex-col items-center animate-wave"
+          :style="{
+            animationDelay: index * 0.05 + 's',
+            '--wave-height': '20px',
+          }"
+        >
+          <img
+            :src="character.src"
+            :alt="character.name"
+            class="w-12 h-12 object-contain drop-shadow-lg transition-transform duration-300"
+          />
+        </div>
+      </div>
+
+      <!-- 로딩 스피너 -->
+      <div class="text-center mb-10">
+        <!-- 멀티 서클 스피너 -->
+        <div class="flex justify-center items-center space-x-2">
+          <div
+            class="animate-bounce-delay-0 w-3 h-3 bg-limegreen-600 rounded-full"
+          ></div>
+          <div
+            class="animate-bounce-delay-1 w-3 h-3 bg-limegreen-500 rounded-full"
+          ></div>
+          <div
+            class="animate-bounce-delay-2 w-3 h-3 bg-limegreen-400 rounded-full"
+          ></div>
+          <div
+            class="animate-bounce-delay-3 w-3 h-3 bg-limegreen-500 rounded-full"
+          ></div>
+          <div
+            class="animate-bounce-delay-4 w-3 h-3 bg-limegreen-600 rounded-full"
+          ></div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup>
+import { onUnmounted, ref, watch } from 'vue';
+
+// 캐릭터 이미지 import
+import characterA from '@/assets/img/characters/A.png';
+import characterB from '@/assets/img/characters/B.png';
+import characterC from '@/assets/img/characters/C.png';
+import characterD from '@/assets/img/characters/D.png';
+import characterE from '@/assets/img/characters/E.png';
+
+// Props
+const props = defineProps({
+  isLoading: {
+    type: Boolean,
+    default: true,
+  },
+  autoClose: {
+    type: Boolean,
+    default: true,
+  },
+  duration: {
+    type: Number,
+    default: 2000, // 기본 2초 (20 * 100ms)
+  },
+  allowOverlayClose: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+const emit = defineEmits(['loading-complete', 'close']);
+
+// 캐릭터 데이터 - 모든 캐릭터가 항상 보임
+const characters = ref([
+  { src: characterA },
+  { src: characterB },
+  { src: characterC },
+  { src: characterD },
+  { src: characterE },
+]);
+
+let loadingTimeout;
+
+// 로딩 시작
+const startLoading = () => {
+  if (props.autoClose) {
+    // duration 시간 후 자동 완료
+    loadingTimeout = setTimeout(() => {
+      emit('loading-complete');
+      emit('close');
+    }, props.duration);
+  }
+};
+
+// 로딩 정지
+const stopLoading = () => {
+  if (loadingTimeout) {
+    clearTimeout(loadingTimeout);
+    loadingTimeout = null;
+  }
+};
+
+// 오버레이 클릭 처리
+const handleOverlayClick = () => {
+  if (props.allowOverlayClose) {
+    stopLoading();
+    emit('close');
+  }
+};
+
+// 로딩 상태가 변경될 때 처리
+watch(
+  () => props.isLoading,
+  newValue => {
+    if (newValue) {
+      startLoading();
+    } else {
+      stopLoading();
+    }
+  }
+);
```

Comment:
```suggestion
watch(
  () => props.isLoading,
  newValue => {
    if (newValue) {
      startLoading();
    } else {
      stopLoading();
    }
  },
  { immediate: true }
);
```
watch를 사용하면 기본적으로 “값이 변할 때만” 동작하기 때문에 처음 컴포넌트가 마운트될 때 props.isLoading이 이미 true인 경우 반응하지 않을 수 있다고 합니다! 
`immediate: true`를 쓰면 처음 마운트될 때도 watch 콜백을 바로 실행할 수 있어서, 초기 로딩 상태를 놓치지 않을 수 있다는데 한 번 고려해보시면 좋을 것 같습니당

Reply:
이부분은 몰랐던 부분인데 말씀해주신것처럼 초기 상태가 true라서 반응하지 않을 수 있기 때문에 적용했습니다!!
감사합니다!
[17d04c5](https://github.com/kb-jjukkumi/choogoomoneyna_FE/pull/63/commits/17d04c54dfb18aa3e93662cdf1af1223bfa415f6)